beesdoo_website_shift/controllers/main.py

```diff
@@ -1,56 +1,437 @@
+        # according to the regular_next_shift_limit
+        if self.is_user_regular() and subscribed_shifts and main_shift:
+            # Get config
+            regular_next_shift_limit = int(request.env['ir.config_parameter'].get_param(
+                'beesdoo_website_shift.regular_next_shift_limit'))
+            for i in range(1, regular_next_shift_limit):
+                # Compute the new date for the created shift
+                start_time = fields.Datetime.from_string(main_shift_rec.start_time)
+                start_time = (start_time + timedelta(days=i*PERIOD)).strftime(DATETIME_FORMAT)
+                # Create the fictive shift
+                shift = copy(main_shift)
+                shift.id = -i  # We give negative id 'caus this shift doesn't exist in database
+                shift.start_day = start_time
+                shift.start_date = start_time
+                subscribed_shifts.append(shift)
+
+        return {
+            'is_regular': self.is_user_regular(),
+            'subscribed_shifts': subscribed_shifts,
+        }
+
+    def my_shift_past_shifts(self):
+        """
+        Return template variables for 'beesdoo_website_shift.my_shift_past_shifts' template
+        """
+        # Get current user
+        cur_user = request.env['res.users'].browse(request.uid)
+        # Get config
+        past_shift_limit = 0
+        if self.is_user_irregular():
+            past_shift_limit = int(request.env['ir.config_parameter'].get_param(
+                'beesdoo_website_shift.irregular_past_shift_limit'))
+        if self.is_user_regular():
+            past_shift_limit = int(request.env['ir.config_parameter'].get_param(
+                'beesdoo_website_shift.regular_past_shift_limit'))
+        # Get shifts where user was subscribed
+        now = datetime.now()
+        if past_shift_limit > 0:
+            past_shifts = request.env['beesdoo.shift.shift'].sudo().search(
+                [('start_time', '<=', now.strftime("%Y-%m-%d %H:%M:%S")),
+                 ('worker_id', '=', cur_user.partner_id.id)],
+                order="start_time desc, task_template_id, task_type_id",
+                limit=past_shift_limit,
+            )
+        else:
+            past_shifts = request.env['beesdoo.shift.shift'].sudo().search(
+                [('start_time', '<=', now.strftime("%Y-%m-%d %H:%M:%S")),
+                 ('worker_id', '=', cur_user.partner_id.id)],
+                order="start_time desc, task_template_id, task_type_id",
+            )
+
+        return {
+            'past_shifts': past_shifts,
+        }
+
+    def my_shift_worker_status(self):
+        """
+        Return template variables for 'beesdoo_website_shift.my_shift_worker_status_*' template
+        """
+        cur_user = request.env['res.users'].browse(request.uid)
+        return {
+            'status': cur_user.partner_id.cooperative_status_ids,
+        }
+
+
+class Shift(object):
```

Comment:
This class was added has I don't find a way to create records of beesdoo.shift.shift that are not saved in the database.

Maybe formatting the date in this class isn't a good idea as it can be performed in the template. But is simplify the template code.

Reply:
you can use rec = ....env['beesdoo.shift.shift'].new()
rec.field1 = 

This class doesn't seems a good idea. the purpose of the template is to format the data.
If you need some specific formatting don't forget that you can use every method of the object shift. Just define the method that return what you need in a class that inherit beesdoo.shift.shift

Reply:
Ok, I will improve that in another PR.